net/api/test_server.cc.h

```diff
@@ -9,17 +9,216 @@
+        Connection& c = r.connection.RawConnection();
+        // Send in some extra, RFC-allowed, whitespaces.
+        c.BlockingWrite("HTTP/1.1  \t  \t\t  200\t    \t\t\t\t  OK\r\n");
+        c.BlockingWrite("Transfer-Encoding: chunked\r\n");
+        c.BlockingWrite("Content-Type: application/octet-stream\r\n");
+        c.BlockingWrite("\r\n");
+        std::this_thread::sleep_for(std::chrono::milliseconds(kDelayBetweenChunksInMilliseconds));
+        for (size_t i = 0; i < numbytes; ++i) {
+          c.BlockingWrite("1\r\n*\r\n");
+          std::this_thread::sleep_for(std::chrono::milliseconds(kDelayBetweenChunksInMilliseconds));
+        }
+        c.BlockingWrite("0\r\n\r\n");  // Line ending as httpbin.org seems to do it. -- D.K.
+        c.SendEOF();
+      };
+      routes[Method::GET]["/status/403"] =
+          [](Request&& r) { r.connection.SendHTTPResponse("", HTTPResponseCode::Forbidden); };
+      routes[Method::GET]["/get"] =
+          [](Request&& r) { r.connection.SendHTTPResponse("{\"Aloha\": \"" + r.url.query["Aloha"] + "\"}\n"); };
+      routes[Method::GET]["/user-agent"] = [](Request&& r) {
+        // TODO(dkorolev): Add parsing User-Agent to Bricks' HTTP headers parser.
+        // The response should be taken from the header, not hardcoded to be "Aloha User Agent".
+        r.connection.SendHTTPResponse("Aloha User Agent");
+      };
+      routes[Method::GET]["/redirect-to"] = [](Request&& r) {
+        HTTPHeadersType headers;
+        headers.push_back(std::make_pair("Location", r.url.query["url"]));
+        r.connection.SendHTTPResponse("", HTTPResponseCode::Found, "text/html", headers);
+      };
+      routes[Method::GET]["/redirect-loop"] = [](Request&& r) {
+        HTTPHeadersType headers;
+        headers.push_back(std::make_pair("Location", "/redirect-loop-2"));
+        r.connection.SendHTTPResponse("", HTTPResponseCode::Found, "text/html", headers);
+      };
+      routes[Method::GET]["/redirect-loop-2"] = [](Request&& r) {
+        HTTPHeadersType headers;
+        headers.push_back(std::make_pair("Location", "/redirect-loop-3"));
+        r.connection.SendHTTPResponse("", HTTPResponseCode::Found, "text/html", headers);
+      };
+      routes[Method::GET]["/redirect-loop-3"] = [](Request&& r) {
+        HTTPHeadersType headers;
+        headers.push_back(std::make_pair("Location", "/redirect-loop"));
+        r.connection.SendHTTPResponse("", HTTPResponseCode::Found, "text/html", headers);
+      };
+      routes[Method::POST]["/post"] = [](Request&& r) {
+        ASSERT_TRUE(r.message.HasBody());
+        r.connection.SendHTTPResponse("{\"data\": \"" + r.message.Body() + "\"}\n");
+      };
+      return routes;
+    }
+
+    bool time_to_terminate_ = false;
+    Server server_;
+  };
+
+  class UniqueImpl {
+   public:
+    UniqueImpl(int port) : port_(port), impl_(new Impl(port)) {}
+    std::string BaseURL() const { return std::string("http://localhost:") + std::to_string(port_); }
```

Comment:
It's better to return or wrap the return value with a URL object which copes with the default port.